### DISCUSSION TITLE
How long does it take to implement features in PrairieLearn?

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#10020] [General]
When writing grants, proposals, and asking for money from institutions to implement features in PrairieLearn, it is often helpful to have a rough idea of how much money to ask for, so here I thought I'd start cataloguing features/implementations and how much they cost.

I'll enumerate things here, and others are welcome to add their experiences and estimates. It would also be helpful to know how much time it took the core devs to review, but maybe that's a separate thing.

| Feature | Feature Size | Devs | Total dev hours | Cost |
| --------| ----- | ----------------| -------------|------|
| #10017 | Small | @knzivid | ~ 4 | $160 CAD |
| #9900 | Large | @knzivid | ~ 45 | $1800 CAD |

For feature size, options are: Small (fix/feature to existing code), Medium, Large, Mega (Manual Grading)

**Comment by Victorsss-Orz** [comment]
| Feature | Feature Size | Devs | Total dev hours | Cost |
| -------- | ------- | ----- | ------ | ------- |
| #9887 | Large | @Victorsss-Orz | ~50-60 | So far 0 (not getting paid yet in the summer) |
